examples/kitchen-sink.yaml

```diff
@@ -379,3 +379,13 @@ resource:
     service.name: !!str "unknown_service"
   # Configure the resource schema URL.
   schema_url: https://opentelemetry.io/schemas/1.16.0
+  # Configure disabled resource attribute keys provided by resource detectors.
+  #
+  # Attribute keys are evaluated to match disabled_attribute_keys in the following manner:
+  #   * If the value of the attribute key exactly matches.
+  #   * If the value of the attribute key matches the wildcard pattern, where '?' matches any single character and '*' matches any number of characters including none.
+  disabled_attribute_keys:
```

Comment:
Maybe this could be an `included`/`excluded` list under `attributes`.

Reply:
Yeah I like that.

Reply:
Scratch that. `.resource.attributes` is currently a map with key values which directly reflect the resource attributes. Its odd to have special-case `included` and `excluded` keys which have values following different rules. 

Lets nest `included` and `excluded` under something which indicates that we're referring to attributes from [resource detectors](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/resource/sdk.md#detecting-resource-information-from-the-environment). Maybe something like:
```
resource:
  detectors:
    attributes:
      included:
        - foo*
        - b?r
      excluded:
        - baz
``` 
Nesting `included` and `excluded` under `.resource.detectors.attributes` because:
- We might want to configure other things related to resource detectors
- If `included` and `excluded` are directly under `.resource.detectors`, then its not clear that we're including / excluding attributes vs. detectors themselves

Reply:
Pushed a commit which reflect this. ^^